src/template.ts

```diff
@@ -1,97 +1,112 @@
-const makeInteger = require('makeInteger');
-const getType = require('getType');
-const Math = require('Math');
-const Object = require('Object');
+const makeInteger: GtmMakeInteger = require('makeInteger');
+const getType: GtmGetType = require('getType');
+const Math: GtmMath = require('Math');
+const Object: GtmObject = require('Object');
 
-const TEMPLATE_VERSION = '1.6.6';
 const INSIGHTS_OBJECT_NAME = 'AlgoliaAnalyticsObject';
-const INSIGHTS_LIBRARY_URL =
-  'https://cdn.jsdelivr.net/npm/search-insights@2.14.0';
+const INSIGHTS_LIBRARY_URL = `https://cdn.jsdelivr.net/npm/search-insights@${devDependencies['search-insights']}`;
 
 const MAX_OBJECT_IDS = 20;
 const MAX_FILTERS = 10;
 const COMMA_REPLACEMENT = 'PRESERVED_COMMA_HERE';
 
-const aa = createArgumentsQueue('aa', 'aa.queue');
+const aa: InsightsClient = createArgumentsQueue('aa', 'aa.queue');
 
 function isInitialized() {
   return !!copyFromWindow(INSIGHTS_OBJECT_NAME);
 }
 
-function replaceEscapedCommas(str) {
+function replaceEscapedCommas(str: string) {
   while (str.indexOf('\\,') !== -1) {
     str = str.replace('\\,', COMMA_REPLACEMENT);
   }
   return str;
 }
 
-function replacePlaceholderText(str) {
+function replacePlaceholderText(str: string) {
   while (str.indexOf(COMMA_REPLACEMENT) !== -1) {
     str = str.replace(COMMA_REPLACEMENT, ',');
   }
   return str;
 }
 
-function splitPreservingEscapedCommas(str) {
+function splitPreservingEscapedCommas(str: string) {
   // GTM does not support regexes in custom templates, so we need to use straight text replacements instead
   return replaceEscapedCommas(str).split(',').map(replacePlaceholderText);
 }
 
-function formatValueToList(value) {
-  const type = getType(value);
-  if (type === 'array') {
+function formatValueToList<T>(value: T | T[]): T[] {
+  if (isArray(value)) {
     return value;
-  } else if (type === 'number') {
+  } else if (isNumber(value)) {
     return [value];
-  } else if (type === 'string') {
-    return splitPreservingEscapedCommas(value);
+  } else if (isString(value)) {
+    return splitPreservingEscapedCommas(value) as T[];
   } else {
-    return null;
+    return [];
   }
 }
 
-function isNullOrUndefined(value) {
+function isArray(value: any): value is any[] {
+  return getType(value) === 'array';
+}
+
+function isNumber(value: any): value is number {
+  return getType(value) === 'number';
+}
+
+function isString(value: any): value is string {
+  return getType(value) === 'string';
+}
+
+function isObject(value: any): value is Record<string, any> {
+  return getType(value) === 'object';
+}
+
+function isNullOrUndefined(value: any): value is null | undefined {
   const type = getType(value);
   return type === 'null' || type === 'undefined';
 }
 
-function transformObjectData(objectData) {
-  const objectDataType = getType(objectData);
-  if (isNullOrUndefined(objectDataType)) {
+function transformObjectData(objectData: any) {
+  if (isNullOrUndefined(objectData)) {
```

Comment:
This is the second actual change.

This was previously incorrect as `objectDataType` would always be a string and then `isNullOrUndefined(objectDataType)` would always return `false`, allowing `objectData` with values of `null` or `undefined` through.